.github/workflows/on-push.yml

```diff
@@ -0,0 +1,81 @@
+# https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Push Verifications
+
+on: [push]
```

Comment:
I just found that this will only run *after* merging pull requests.

I think we want something more like this. The syntax is a little awkward, but it will trigger the job on all pull requests, pushes to the master branch, and releases.
```
on:
  pull_request:
  push:
    branches:
      - master
  release:
```

Reply:
Would the sub-specification for push of `branches: master` prevent the workflow from running in the user's non-master branch (e.g. before submitting as a PR)?  Because right now it works fine in my local branch, it just doesn't show up in the PR.

Reply:
> prevent the workflow from running in the user's non-master branch

Yes that is my understanding. If we change to `[push, pull_request]`, the tests will run on a user's non-master branch, but we'll get duplicate jobs for everything after a PR is made.

Reply:
Hmm, so I tried just adding `pull_request` and `release` and leaving `push` alone, but I'm still not seeing it in the PR.  I am still seeing the workflow execute in my local branch.

So it's possible instead of duplicating workflow execution, it's only running in my local branch and not in the PR.  I'll add the `branches: master` and see if I see it then show up in the PR.

Reply:
> Hmm, so I tried just adding pull_request and release and leaving push alone, but I'm still not seeing it in the PR. 

I suspect that is because it hasn't landed in the pipx source code yet

Reply:
huh.  I committed your suggestion and am still not seeing it here in this PR.  And also not in my local branch.  I'll investigate.

Reply:
Or maybe we won't see it in the pipx repo until we finally merge the PR?

Reply:
> we won't see it in the pipx repo until we finally merge the PR?

Yes I think this is the case. Or if you make the branch in `pipxproject/github-actions` rather than your personal repo.

Reply:
Do we need to prevent duplicate workflows with the `branches:` specification?  

I'm wondering if github would do the right thing and only execute one workflow but have it show up in two different places.

Mostly I would just be sad to lose the CI feedback when developing in my own local branch. 🙂

Reply:
No we don't need to. It makes things a little noisier, possibly take longer, and use more GitHub resources (which are free for public repos... crazy). None of those are a dealbreaker. 

We can try merging this with all PRs and pushes turned on and go from there.

Reply:
Maybe we can check in the future and attempt to verify if it's really duplicating workflow executions or just displaying the same workflow in different contexts.